### PR TITLE
feat(bridge): integrate storybook

### PR DESCRIPTION
**Pull request by renepanzar**
## This PR

Adds Storybook.js to the bridge so we can develop components independently from their context. It also acts as documentation for the components and best practices.

### Notes

image

### Follow-up Tasks

Best practices will be added later on.

### How to test

1. `yarn storybook` will start the Storybook development server and open the endpoint in a browser
2. Click through the stories

### CODE REVIEW

File: bridge/client/app/_components/ktb-date-input/ktb-date-input.module.ts [hunk 1]
```diff
@@ -1,19 +1,21 @@
...
 import { DtIconModule } from '@dynatrace/barista-components/icon';
+import { DtInputModule } from '@dynatrace/barista-components/input';
+import { KtbDatetimePickerComponent, KtbDatetimePickerDirective } from './ktb-datetime-picker.component';
+import { KtbTimeInputComponent } from './ktb-time-input.component';
 
 @NgModule({
   declarations: [KtbDatetimePickerComponent, KtbTimeInputComponent, KtbDatetimePickerDirective],
   imports: [
     CommonModule,
+    RouterModule,
```
Comment (Kirdock) [inline bridge/client/app/_components/ktb-date-input/ktb-date-input.module.ts:18]: Why do we need the RouterModule here?

File: bridge/client/app/_components/ktb-copy-to-clipboard/ktb-copy-to-clipboard.module.ts [hunk 2]
```diff
@@ -1,16 +1,18 @@
...
-import { DtButtonModule } from '@dynatrace/barista-components/button';
 import { DtInputModule } from '@dynatrace/barista-components/input';
-import { FlexModule } from '@angular/flex-layout';
+import { KtbCopyToClipboardComponent } from './ktb-copy-to-clipboard.component';
 
 @NgModule({
   declarations: [KtbCopyToClipboardComponent],
   imports: [
     CommonModule,
+    HttpClientModule,
```
Comment (Kirdock) [inline bridge/client/app/_components/ktb-copy-to-clipboard/ktb-copy-to-clipboard.module.ts:15]: Could you move that to the imports of the required story instead?

File: bridge/client/app/_components/ktb-app-header/ktb-app-header.module.ts [hunk 3]
```diff
@@ -1,24 +1,26 @@
...
 import { KtbCopyToClipboardModule } from '../ktb-copy-to-clipboard/ktb-copy-to-clipboard.module';
-import { DtIconModule } from '@dynatrace/barista-components/icon';
+import { KtbAppHeaderComponent } from './ktb-app-header.component';
+import { KtbUserComponent } from './ktb-user/ktb-user.component';
 
 @NgModule({
   declarations: [KtbAppHeaderComponent, KtbUserComponent],
   imports: [
     CommonModule,
+    HttpClientModule,
```
Comment (Kirdock) [inline bridge/client/app/_components/ktb-app-header/ktb-app-header.module.ts:23]: Could you move that to the imports of the required story instead?

File: bridge/stories/components/Heatmap.stories.ts [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+  IDataPoint,
+  IHeatmapTooltipType,
+} from '../../client/app/_interfaces/heatmap';
+import { ResultTypes } from '../../shared/models/result-types';
+
+export default {
+  title: 'Components/Heatmap',
+  decorators: [
+    moduleMetadata({
+      imports: [KtbHeatmapModule],
```
Comment (Kirdock) [inline bridge/stories/components/Heatmap.stories.ts:18]: `HttpClientModule` is missing here

File: bridge/stories/components/Components.stories.mdx [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+<Story id="components-selectable-tile--selected" />
+
+## Danger Zone
+
+<Story id="components-danger-zone--project" />
+
+<div>
+  <h1>Blubb</h1>
+</div>
```
Comment (Kirdock) [inline bridge/stories/components/Components.stories.mdx:48]: Could you remove this test data?

**Review by Kirdock — CHANGES_REQUESTED** [review]
There is still a bunch of errors in the console.
e.g.:
image
image

and the certificate-input for example also throws a lot of errors
image


At the moment the project is not set in the header anymore if you refresh the page
image

File: bridge/.gitignore [hunk 6]
```diff
@@ -58,3 +58,4 @@ e2e/screenshots/*
 !cypress/fixtures/files/*.pem
 *.crt
 *.key
+/documentation.json
```
Comment (ermin-muratovic) [inline bridge/.gitignore:61]: It looks like `/documentation.json` belongs to the group `# certificates`.
Could you add a new comment `# storybook` and move this below that.

**Review by ermin-muratovic — CHANGES_REQUESTED** [review]

**Review by ermin-muratovic — APPROVED** [review]
lgtm

**Review by Kirdock — CHANGES_REQUESTED** [review]
Some components still throw some errors.
- [x] Certificate input: When you manually change the input:
image
- [x] Expandable tile: The tile immediately closes for the "Expanded" story.
- [x] App Header: There is an error thrown if you select a project

- [ ] I guess the global styles.scss is missing here. That would explain some incorrect styles, like for the copy-clipboard icon

**Comment by renepanzar** [comment]
Thanks, @Kirdock, for your comments.

I've fixed the certificate input issues.

The expandable tile does have an issue, though, it doesn't want to stay opened. This needs to be fixed inside the component.

The app-header does not produce an error any longer with missing route.

The icon in the clipboard component needs to be re-styled, it's an issue with the component itself. The global styles are loaded, or else it would look like the following (app-header):

image

**Comment by renepanzar** [comment]
Out of curiosity I tried to fix the expandable tile, which now works.

File: bridge/.storybook/tsconfig.json [hunk 7]
```diff
@@ -0,0 +1,10 @@
+{
+  "extends": "../tsconfig.app.json",
+  "compilerOptions": {
+    "types": ["node"],
+    "allowSyntheticDefaultImports": true
+  },
+  "exclude": ["../src/test.ts", "../src/**/*.spec.ts", "../projects/**/*.spec.ts"],
+  "include": ["../src/**/*", "../projects/**/*"],
```
Comment (Kirdock) [inline bridge/.storybook/tsconfig.json:8]: Those paths do not exist.

**Review by Kirdock — CHANGES_REQUESTED** [review]

File: bridge/.storybook/tsconfig.json (hunk 7 above)
Comment (Kirdock) [reply bridge/.storybook/tsconfig.json:8]: Same for `tsconfig.doc.json`

**Review by Kirdock — APPROVED** [review]

**Pull request merged**